OHS-Hosting-Infrastructure/complaint-tracker
# Refactor of Api to make it flexible for future

## Description of change
This change is a proposal! 

The current structure we have for the Api response is not very OO, and will not scale very well. I have done a preliminary refactor that I believe will be more flexible going forward.

I propose having FakeData and real Api classes that mirror each other's structure so that we can do a basic switch on environments to access the real data (when that is available) in staging and production and the fake data when we are testing or in development. This will also give us a lot of flexibility in the real API to switch from staging to production endpoints and credentials.

I also switched the FakeData caching to a Singleton -- that is a little cleaner than a global variable as it has to be required into a file, and also will allow data to persist between app restarts in a little more realistic way.

If this proposal meets everyone's approval, I will create an ADR to describe why we took this approach.

## Acceptance Criteria

## How to test
Click around on the site and make sure the index view and detail view still work.


## Checklist

To be completed by the submitter:



## To the Reviewer

This project is using [Conventional Comments](https://conventionalcomments.org/) to give structure
and context to PR comments. Please use these labels in your comments.

* **praise:**
* **nitpick:**
* **suggestion:**
* **issue:**
* **question:**
* **thought:**
* **chore:**
